lib/modules/collapseWhitespace.es6

```diff
@@ -19,8 +32,27 @@ export default function collapseWhitespace(tree, options, collapseType, tag) {
 
     tree.forEach((node, index) => {
         if (typeof node === 'string' && !isComment(node)) {
-            const isTopLevel = ! tag || tag === 'html' || tag === 'head';
-            node = collapseRedundantWhitespaces(node, collapseType, isTopLevel);
+            const prevNode = tree[index - 1];
+            const nextNode = tree[index + 1];
+            const prevNodeTag = prevNode && prevNode.tag;
+            const nextNodeTag = nextNode && nextNode.tag;
+
+            const isTopLevel = !tag || tag === 'html' || tag === 'head';
+
+            const shouldTrim = (
+                collapseType === 'all' ||
+                isTopLevel ||
+                /*
+                 * When collapseType is set to 'aggressive', and the tag is not inside 'noTrimWhitespacesInsideElements'.
+                 * the first & last space inside the tag will be trimmed
+                 */
+                (
+                    collapseType === 'aggressive' &&
+                    !noTrimWhitespacesInsideElements.has(tag)
+                )
```

Comment:
```
    if (collapseType === 'all' || isTopLevel) {	
```

Move the original `trim` logic here as a `shouldTrim`, along with `! noTrimWhitespacesInsideElements.has(tag)` to trim even more spaces when `collapseType` is set to `aggressive`.